docs/memory_layout.md

```diff
@@ -0,0 +1,230 @@
+| mat2x4\<float>[] | <kbd>[bbbb][bbbb][bbbb][bbbb]<br/>[bbbb][bbbb][bbbb][bbbb]                                                          </kbd> |
+| mat3x2\<float>[] | <kbd>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]                             </kbd> |
+| mat4x2\<float>[] | <kbd>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]</kbd> |
+| mat4x3\<float>[] | <kbd>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]</kbd> |
+
+### STD430
+| Name             | Bytes                                                                                                                      |
+|------------------|----------------------------------------------------------------------------------------------------------------------------|
+| mat2x2\<fp16>[]  | <kbd>[bbbb][bbbb]                                                                                                   </kbd> |
+| mat2x2\<float>[] | <kbd>[bbbb][bbbb][bbbb][bbbb]                                                                                       </kbd> |
+| mat2x3\<float>[] | <kbd>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]                                                          </kbd> |
+| mat2x4\<float>[] | <kbd>[bbbb][bbbb][bbbb][bbbb]<br/>[bbbb][bbbb][bbbb][bbbb]                                                          </kbd> |
+| mat2x2\<int8>[]  | <kbd>[bbbb]                                                                                                         </kbd> |
+| mat3x2\<float>[] | <kbd>[bbbb][bbbb][bbbb][bbbb]<br/>[bbbb][bbbb]                                                                      </kbd> |
+| mat3x3\<float>[] | <kbd>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]                             </kbd> |
+| mat4x2\<float>[] | <kbd>[bbbb][bbbb][bbbb][bbbb]<br/>[bbbb][bbbb][bbbb][bbbb]                                                          </kbd> |
+| mat4x3\<float>[] | <kbd>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]</kbd> |
+
+## Structures
+```
+struct {
+  int32 w;
+  float x;
+}
+```
+
+The STD140 pads 8 bytes at the end to become a multiple of 16 bytes.
+
+| STD | Stride | Bytes                                                                    |
+|-----|:------:|--------------------------------------------------------------------------|
+| 140 |   16   | <kbd>{w [bbbb]}<br/>{x [bbbb]}<br/> [----][----]           </kbd> |
+| 430 |    8   | <kbd>{w [bbbb]}<br/>{x [bbbb]}                                    </kbd> |
+
+
+<hr>
+
+```
+struct {
+  int32 w;
+  vec2<float> x;
+  float y;
+}
+```
+
+For both cases, the `vec2` is the largest item, so giving a base alignment of
```

Comment:
"largest item" -> "member with the largest base alignment requirement"  (to be perfectly explicit about it)

Reply:
done